scripts/o3de/o3de/export_project.py

```diff
@@ -136,3 +283,402 @@ def add_args(subparsers) -> None:
+    """
+    Build (or rebuild) the export tool chain (AssetProcessorBatch and AssetBundlerBatch)
+
+    @param ctx:                 Export Context
+    @param tools_build_path:    The tools (cmake) build path to create the build project for the tools
+    @param logger:              Optional Logger
+    @return: None
+    """
+
+    # Generate the project for export toolchain
+    cmake_configure_command = ["cmake", "-B", tools_build_path]
+
+    if ctx.is_engine_centric:
+        cmake_configure_command.extend(["-S", ctx.engine_path])
+    else:
+        cmake_configure_command.extend(["-S", ctx.project_path])
+
+    # Generate the project for the pre-requisite tools
+    if GENERATOR:
+        cmake_configure_command.extend(["-G", GENERATOR])
+    if CMAKE_GENERATOR_OPTIONS:
+        cmake_configure_command.extend(CMAKE_GENERATOR_OPTIONS)
+    if not CMAKE_MULTI_CONFIGURATION_GENERATOR:
+        cmake_configure_command.extend([f'-DCMAKE_BUILD_TYPE={PREREQUISITE_TOOL_BUILD_CONFIG}'])
+    if ctx.is_engine_centric:
+        cmake_configure_command.extend([f'-DLY_PROJECTS={ctx.project_path.name}'])
+    if logger:
+        logger.info(f"Generating tool chain files for project {ctx.project_name}.")
+    ret = process_command(cmake_configure_command)
+    if ret != 0:
+        raise ExportProjectError("Error generating the project for the pre-requisite tools.")
+
+    # Build the project for the pre-requisite tools
+    cmake_build_command = ["cmake", "--build", tools_build_path]
+
+    if CMAKE_MULTI_CONFIGURATION_GENERATOR:
+        cmake_build_command.extend(["--config", PREREQUISITE_TOOL_BUILD_CONFIG])
+
+    cmake_build_command.extend(["--target", "AssetProcessorBatch", "AssetBundlerBatch"])
+
+    if ctx.cmake_additional_build_args:
+        cmake_build_command.extend(ctx.cmake_additional_build_args)
+    if logger:
+        logger.info(f"Building tool chain files for project {ctx.project_name}.")
+    ret = process_command(cmake_build_command)
+    if ret != 0:
+        raise ExportProjectError("Error building the project for the pre-requisite tools.")
+    if logger:
+        logger.info(f"Tool chain built successfully.")
+
+
+def validate_export_toolchain(tools_build_path: pathlib.Path):
+    """
+    Validate that the required command line tools are available for the export process
+
+    @param tools_build_path:    The base tools build path
+    """
+    # Otherwise make sure the tools exist already
+    build_tools = [get_asset_bundler_batch_path(tools_build_path),
+                   get_asset_processor_batch_path(tools_build_path)]
+    tools_missing = [b for b in build_tools if not b.exists()]
+    if len(tools_missing) > 0:
+        raise ExportProjectError(f"Necessary Build Tools have not been created! The following are missing: {', '.join(tools_missing)}"
+                                 "Please ensure that these tools exist before proceeding on with the build!")
```

Comment:
I would only throw an exception if these tools need to be run.
If user has already built and bundled their assets and only need to re-build the launchers, then this doesn't need to error.

Reply:
Right now the there is a 'should_build_all_assets' which should probably need to change to 'should_build_asset_and_bundle' or 'should_build_bundle' since building the assets should be implied if you want to rebuild the bundle. I would have to make this change and rely on the flag to check on the tools

Reply:
There was no real need for this function, since we want a more granular check, i will check each tool if needed